book-cw/part-2-coding/regex.adoc

```diff
@@ -1,3 +1,348 @@
 = Coding in Regex
 
-Chapter content removed for re-review (in progress).
+The patterning syntax known as _regular expressions_ or _regex_ is as notorious for its alleged steep learning curve as for its confirmed usefulness.
```

Comment:
Before you tell me that regex is difficult, tell me what it's useful for, to get me psyched up to deal with cryptic syntax because it's worth it.

Reply:
Yes, I'd start off with a simple but realistic example of a regex and what it can do. Maybe something like

  newText = text.replace(/word/g, 'new word');

to replace instances of 'word' everywhere with 'new word'.

Then point out that regex is more than just substring matching, it also knows about things like word boundaries. So you could make that last expression a bit more robust (so you don't end up with things like "sword" transforming to "snew word").

  newText = text.replace(/\bword\b/g, 'new word');

You don't want to go too far down this road in the beginning, of course, but at least it gives a hint of the power without just saying it has a notoriously steep learning curve.